source/PlayServicesResolver/src/PlayServicesResolver.cs

```diff
@@ -928,32 +926,69 @@ public class AndroidSdkRootChangedArgs : EventArgs {
         /// Defaults to 1 second.</param>
         private static void ScheduleAutoResolve(double delayInMilliseconds = 1000.0) {
             lock (typeof(PlayServicesResolver)) {
-                if (!autoResolving) {
-                    RunOnMainThread.Cancel(autoResolveJobId);
-                    autoResolveJobId = RunOnMainThread.Schedule(
-                        () => {
-                            lock (typeof(PlayServicesResolver)) {
-                                autoResolving = true;
-                            }
-                            RunOnMainThread.PollOnUpdateUntilComplete(() => {
-                                if (EditorApplication.isCompiling) return false;
-                                // Only run AutoResolve() if we have a valid autoResolveJobId.
-                                // If autoResolveJobId is 0, ScheduleResolve()
-                                // has already been run and we should not run AutoResolve()
-                                // again.
-                                if (autoResolveJobId != 0) {
-                                    AutoResolve(() => {
-                                        lock (typeof(PlayServicesResolver)) {
-                                            autoResolving = false;
-                                            autoResolveJobId = 0;
+                if (autoResolving)
+					return;
+
+				RunOnMainThread.Cancel(autoResolveJobId);
+                autoResolveJobId = RunOnMainThread.Schedule(() => {
+                    lock (typeof(PlayServicesResolver)) {
+                        autoResolving = true;
+                    }
+
+                    DateTimeOffset delay = DateTimeOffset.Now.AddSeconds(Math.Min(30,GooglePlayServices.SettingsDialog.AutoResolutionDelay));
+                    bool shouldResolve = true;
+                    AlertModal alert = null;
+                    RunOnMainThread.PollOnUpdateUntilComplete(() => {
```

Comment:
What happens if a user uses the menu to start "Resolve" or "Force Resolve" when this job is running?

Typically the auto-resolve job is canceled here https://github.com/googlesamples/unity-jar-resolver/blob/master/source/PlayServicesResolver/src/PlayServicesResolver.cs#L1397 since the job Id isn't saved (it would also need to be in a critical section) this could result in multiple resolve jobs running at the same time.

Reply:
Doesn't that would be guarded at here? https://github.com/googlesamples/unity-jar-resolver/blob/e703601cec96ff60883b8999ca9aaa3ef2e459d9/source/PlayServicesResolver/src/PlayServicesResolver.cs#L944

I think if there would be any resolution interrupted, it would set `autoResolveJobId` to 0 at 
https://github.com/googlesamples/unity-jar-resolver/blob/e703601cec96ff60883b8999ca9aaa3ef2e459d9/source/PlayServicesResolver/src/PlayServicesResolver.cs#L1398 so if user use force resolve before the dialog was closed this job it would just be skip even user press OK